Fix last issues of follow mode and remove feature toggle

tobous opened the pull request:
This PR is grouped into multiple commits. Each commit is dealing with a specific issue and should be reviewed separately. Most commits contain a commit message giving further information about their purpose or the decisions made during their creation.

tobous commented on de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorStatusChangeHandler.java:
```diff
@@ -1,5 +1,6 @@
 package de.fu_berlin.inf.dpp.intellij.editor;
```
As stated in the commit message, the logic added to this handler will be moved into a separate handler in an upcoming commit that moves all the Intellij event handler into a separate sub-package.

tobous commented on de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorStatusChangeHandler.java:
```diff
@@ -206,4 +264,106 @@ public void setEnabled(boolean enabled) {
+    }
+  }
+
+  /** Data storage class for queued viewport changes. */
```
I am not sure whether this data storage class should be moved to a separate class. Any opinions?

m273d15 replied:
Can remain if you remove the `toString` method.

tobous commented on de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorStatusChangeHandler.java:
```diff
@@ -206,4 +264,106 @@ public void setEnabled(boolean enabled) {
+    if (previouslyQueuedRequest != null) {
+      if (log.isTraceEnabled()) {
+        log.trace(
+            "Dropping previously queued viewport change request for "
```
I am not sure whether this log is really necessary. Especially because it introduces a log of boilerplate code with the `QueuedViewPortChange.toString()` method. Any opinions?

tobous commented:
I hope you are not discouraged by the size of this PR. Most of the changes happen in `[FIX][I] Queue viewport adjustments if editor is not currently visible`. The rest are mainly minor bugfixes. The large commit also doesn't add that much complicated logic. It also contains a lot of boilerplate stuff.

m273d15 commented on de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/actions/FollowModeAction.java:
```diff
@@ -52,7 +52,13 @@ public void execute(String userName) {
     }
 
-    currentFollowModeManager.follow(userToFollow);
+    User currentlyFollowedUser = followModeManager.getFollowedUser();
```
I would prefer to move this logic into the `FollowModeManager`

m273d15 commented on de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorManager.java:
```diff
@@ -852,21 +861,46 @@ public void closeEditor(final SPath path) {
+      return;
+    }
+
+    ApplicationManager.getApplication()
```
Either use `executeInUIThreadSynchronous` or `ApplicationManager...` directly and remove wrapper.

tobous commented:
Moved logic to avoid unnecessary follow mode changes to FollowModeManager (now in #390).

Removed queued viewport change replacement logging and `toString`.

User `executeInUIThreadSynchronous` instead of `ApplicationManager.getApplication().invokeAndWait()`.

tobous commented:
Rebase onto current master without any interaction.

tobous commented:
Did some force pushes to trigger Travis PR build correctly.
